src/app/Console/Commands/RequireDevTools.php

```diff
@@ -35,63 +35,71 @@ class RequireDevTools extends Command
      */
     public function handle()
     {
-        $this->progressBar = $this->output->createProgressBar(29);
+        $this->progressBar = $this->output->createProgressBar(45);
         $this->progressBar->minSecondsBetweenRedraws(0);
         $this->progressBar->maxSecondsBetweenRedraws(120);
         $this->progressBar->setRedrawFrequency(1);
-
         $this->progressBar->start();
 
         $this->info(' Requiring DevTools. Please wait...');
         $this->progressBar->advance();
 
-        // Check if auth exists
-        $details = null;
-        $process = new Process(['composer', 'config', 'http-basic.backpackforlaravel.com']);
-        $process->run(function ($type, $buffer) use (&$details) {
-            if ($type !== Process::ERR && $buffer !== '') {
-                $details = json_decode($buffer);
-            } elseif (File::exists('auth.json')) {
-                $details = json_decode(File::get('auth.json'), true)['http-basic']['backpackforlaravel.com'] ?? false;
-            }
-            $this->progressBar->advance();
-        });
+        // Check if repositories exists
+        $this->composerRepositories();
 
-        // Create an auth.json file
-        if (! $details) {
-            $this->info(' Creating auth.json file with your authentication token');
+        // Check for authentication
+        $this->checkForAuthentication();
 
-            $this->line(' (Find your access token details on https://backpackforlaravel.com/user/tokens)');
-            $username = $this->ask('Access token username');
-            $password = $this->ask('Access token password');
+        // Require DevTools
+        $this->requireDevTools();
 
-            $process = new Process(['composer', 'config', 'http-basic.backpackforlaravel.com', $username, $password]);
-            $process->run(function ($type, $buffer) use ($username, $password) {
-                if ($type === Process::ERR) {
-                    // Fallback
-                    $authFile = [
-                        'http-basic' => [
-                            'backpackforlaravel.com' => [
-                                'username' => $username,
-                                'password' => $password,
-                            ],
-                        ],
-                    ];
+        // Check if devtools is installed
+        $tries = 2;
```

Comment:
The script will ask for user credentials 2 times, after the second wrong attempt it will fail.